core/extension/gdextension_interface.h

```diff
@@ -155,27 +155,45 @@ typedef enum {
 // - Some types have no destructor (see `extension_api.json`'s `has_destructor` field), for
 //   them it is always safe to skip the constructor for the return value if you are in a hurry ;-)
 
-typedef void *GDExtensionVariantPtr;
-typedef const void *GDExtensionConstVariantPtr;
-typedef void *GDExtensionUninitializedVariantPtr;
-typedef void *GDExtensionStringNamePtr;
-typedef const void *GDExtensionConstStringNamePtr;
-typedef void *GDExtensionUninitializedStringNamePtr;
-typedef void *GDExtensionStringPtr;
-typedef const void *GDExtensionConstStringPtr;
-typedef void *GDExtensionUninitializedStringPtr;
-typedef void *GDExtensionObjectPtr;
-typedef const void *GDExtensionConstObjectPtr;
-typedef void *GDExtensionUninitializedObjectPtr;
-typedef void *GDExtensionTypePtr;
-typedef const void *GDExtensionConstTypePtr;
-typedef void *GDExtensionUninitializedTypePtr;
-typedef const void *GDExtensionMethodBindPtr;
+#define GDEXTENSION_OPAQUE_PTR(type) \
+	typedef struct GDExtensionOpaque##type *GDExtension##type##Ptr;
+
+#define GDEXTENSION_CONST_OPAQUE_PTR(type) \
+	typedef const struct GDExtensionOpaque##type *GDExtensionConst##type##Ptr;
+
+#define GDEXTENSION_UNINITIALIZED_OPAQUE_PTR(type) \
+	typedef struct GDExtensionUninitialized##type *GDExtensionUninitialized##type##Ptr;
+
+GDEXTENSION_OPAQUE_PTR(Variant)
+GDEXTENSION_CONST_OPAQUE_PTR(Variant)
+GDEXTENSION_UNINITIALIZED_OPAQUE_PTR(Variant)
+
+GDEXTENSION_OPAQUE_PTR(StringName)
+GDEXTENSION_CONST_OPAQUE_PTR(StringName)
+GDEXTENSION_UNINITIALIZED_OPAQUE_PTR(StringName)
+
+GDEXTENSION_OPAQUE_PTR(String)
+GDEXTENSION_CONST_OPAQUE_PTR(String)
+GDEXTENSION_UNINITIALIZED_OPAQUE_PTR(String)
+
+GDEXTENSION_OPAQUE_PTR(Object)
+GDEXTENSION_CONST_OPAQUE_PTR(Object)
+GDEXTENSION_UNINITIALIZED_OPAQUE_PTR(Object)
+
+GDEXTENSION_OPAQUE_PTR(Type)
+GDEXTENSION_CONST_OPAQUE_PTR(Type)
+GDEXTENSION_UNINITIALIZED_OPAQUE_PTR(Type)
+
+GDEXTENSION_CONST_OPAQUE_PTR(MethodBind)
+
 typedef int64_t GDExtensionInt;
 typedef uint8_t GDExtensionBool;
+typedef double GDExtensionFloat;
```

Comment:
I don't think we're using floats in the interface anywhere, and I'm also unsure if we want to always use `double`s for them?